experimental/oak_functions_loader_linux_vsock/src/channel.rs

```diff
@@ -0,0 +1,141 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::proto::oak::sandbox::runtime::{Request, Response};
+use anyhow::anyhow;
+use byteorder::{BigEndian, ByteOrder};
+use prost::Message;
+use ringbuf::{Consumer, Producer, RingBuffer};
+use std::{
+    io::{Cursor, Read, Write},
+    mem::size_of,
+};
+
+const BUFFER_SIZE: usize = 65536;
```

Comment:
This size is probably excessive. The current max size for a frame is 4000 bytes (not 64000 as it used to be).